src/YoloObjectDetector.cpp

```diff
@@ -136,11 +154,30 @@ void YoloObjectDetector::on_image_callback(const cv::Mat& image)
     frameHeight_ = image.rows;
     if (!writer_configured_)
     {
-      std::string url_writer(this->get_namespace());
-      url_writer = "rtsp://127.0.0.1:8554"+url_writer;
-      std::cout << "url_writer: " << url_writer << std::endl;
-      // Configure RTSP Streamer
-      rtsp_streamer_.on_configure_writer(frameWidth_, frameHeight_, 30, 9000, url_writer);
+      // Check whether a GStreamer pipeline has been provided
+      std::string gstreamer_pipeline;
+      get_parameter("gstreamer_writer_pipeline", gstreamer_pipeline);
```

Comment:
Would it be needed to also be  able to set the `reader_pipeline`? Because lets say I write from essentials in a custom way, I need to also read it in here in a custom way isn't it? There might be a way to automatically read it independently of the input. That would be our target. For now thought just consider it as some brainstorming idea

Reply:
It does make sense. I will take a look at how could we do this and come back to you!

Reply:
Ah yes, I remember now. In `umd_streaming`, in `rtsp_streamer.hpp/.cpp` we have a method for opening a **writer** pipeline using as input either the stream URL or a custom pipeline; meanwhile, for opening a **reader** we only have a method that accepts the stream URL as input. Thus, we would need to open a PR for `umd_streaming`, to add a method for `on_configure_reader` that could take a custom pipeline as an input.